lnst/Recipes/ENRT/MeasurementGenerators/XDPFlowMeasurementGenerator.py

```diff
@@ -0,0 +1,30 @@
+from lnst.Common.Parameters import ChoiceParam, StrParam
+from lnst.RecipeCommon.Perf.Measurements.XDPBenchMeasurement import XDPBenchMeasurement
+from lnst.Recipes.ENRT.MeasurementGenerators.BaseFlowMeasurementGenerator import (
+    BaseFlowMeasurementGenerator,
+)
+
+from lnst.Tests.XDPBench import XDP_BENCH_COMMANDS
+
+
+class XDPFlowMeasurementGenerator(BaseFlowMeasurementGenerator):
+    xdp_command = ChoiceParam(type=StrParam, choices=XDP_BENCH_COMMANDS)
+
+    @property
+    def net_perf_tool_class(self):
```

Comment:
so i thought of a ... python magic thing for this:

```python
@property
def net_perf_tool_class(self):
    def XDPBenchMeasurement_partial(*args):
        return XDPBenchMeasurement(*args, self.params.xdp_command)
    return XDPBencMEasurement_partial
```

what this does is it **binds** one of the parameters of the class initializer to a specific value and returns a "new initializer" which takes one less initializer.

This is a concept from functional programming of 'partial application' : https://en.wikipedia.org/wiki/Partial_application

aaaand python actually has a thing for this: https://docs.python.org/3/library/functools.html?highlight=partial#functools.partial

so my previous example maybe can be simplified with that if it works with classes... but maybe it doesn't need to be.

why i mention this... it would basically remove the need for you to reimplement the `generate_perf_measurements_combinations` method just because of this one parameter and so you wouldn't have any issues with the super calls or breaking any sort of collaborative inheritance.

though... this would probably be best to document in a docstring for anyone else in the future looking at this.

Reply:
That's a nice idea, thanks.